elide-core/src/main/java/com/yahoo/elide/security/permissions/ExpressionBuilder.java

```diff
@@ -0,0 +1,203 @@
+ */
+package com.yahoo.elide.security.permissions;
+
+import com.yahoo.elide.audit.InvalidSyntaxException;
+import com.yahoo.elide.core.PersistentResource;
+import com.yahoo.elide.security.ChangeSpec;
+import com.yahoo.elide.security.checks.Check;
+import com.yahoo.elide.security.checks.ExtractedChecks;
+import com.yahoo.elide.security.permissions.expressions.AndExpression;
+import com.yahoo.elide.security.permissions.expressions.AnyFieldExpression;
+import com.yahoo.elide.security.permissions.expressions.DeferredCheckExpression;
+import com.yahoo.elide.security.permissions.expressions.Expression;
+import com.yahoo.elide.security.permissions.expressions.ImmediateCheckExpression;
+import com.yahoo.elide.security.permissions.expressions.OrExpression;
+import com.yahoo.elide.security.permissions.expressions.SpecificFieldExpression;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+import static com.yahoo.elide.security.checks.ExtractedChecks.CheckMode.ALL;
+import static com.yahoo.elide.security.checks.ExtractedChecks.CheckSubset;
+
+import java.lang.annotation.Annotation;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Queue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+/**
+ * Expression builder to parse annotations and express the result as the Expression AST.
+ */
+@Slf4j
+public class ExpressionBuilder {
+    private final HashMap<Check, ExpressionResult> cache = new HashMap<>();
+
+    private static final BiFunction<Expression, Expression, Expression> ALL_JOINER = AndExpression::new;
+    private static final BiFunction<Expression, Expression, Expression> ANY_JOINER = OrExpression::new;
+
+    /**
+     * Build an expression that checks a specific field.
+     *
+     * @param resource Resource
+     * @param annotationClass Annotation calss
+     * @param field Field
+     * @param changeSpec Change spec
+     * @param <A> Type parameter
+     * @return Commit and operation expressions
+     */
+    public <A extends Annotation> Expressions buildSpecificFieldExpression(final PersistentResource resource,
+                                                                           final Class<A> annotationClass,
+                                                                           final String field,
+                                                                           final ChangeSpec changeSpec) {
+        ExtractedChecks extracted = new ExtractedChecks(resource, annotationClass, field);
+
+        Expression operationExpression =
+                buildFullExpression(extracted.getEntityChecks(),
+                                    extracted.getFieldChecks(),
+                                    (check) -> new DeferredCheckExpression(check, resource, changeSpec, cache),
+                                    SpecificFieldExpression::new);
```

Comment:
Instead of passing in to the next function the 'expression joiner' here, it might make more sense to do that final expression joining here?